### PR TITLE
[7.0][native][Machine-learning][ACR-1721] Add new layer description

### PR DESCRIPTION
**Pull request by songgot**
[7.0][native][Machine-learning][ACR-1721] Add new layer description
    
    - Add attention, mol attention, multi head attention, layer normalization
      and positional encoding layer description
    
    Signed-off-by: hyunil park <hyunil46.park@samsung.com>

### CODE REVIEW

File: docs/application/native/guides/machine-learning/machine-learning-training.md [hunk 1]
```diff
@@ -266,8 +266,51 @@ Type | Key | Value | Default value | Description
...
-&#xfeff;                                                     | hidden_state_zoneout_rate   | (float)                     | 0                       | zoneout rate for hidden state
+&#xfeff;                                                     | hidden_state_zoneout_rate   | (float)                     | 0                       | Zoneout rate for hidden state
 &#xfeff;                                                     | integrate_bias              | (boolean)                   | false                   | Integrate bias_ih, bias_hh to bias_h
+`ML_TRAIN_LAYER_TYPE_ATTENTION` (since 7.0)                  |                             |                             |                         | Attention layer
+&#xfeff;                                                     | n/a                         |                             |                         |
+`ML_TRAIN_LAYER_TYPE_MOL_ATTENTION` (since 7.0)              |                             |                             |                         | Mol attention layer
```
Comment (lhs8928) [inline docs/application/native/guides/machine-learning/machine-learning-training.md:273]: Mol => MoL

File: docs/application/native/guides/machine-learning/machine-learning-training.md [hunk 2]
```diff
@@ -266,8 +266,51 @@ Type | Key | Value | Default value | Description
...
+&#xfeff;                                                     | unit                        | (unsigned integer)          |                         | Number of output neurons
+&#xfeff;                                                     | MoL_K                       | (unsigned integer)          |                         | Size of the three projections
+`ML_TRAIN_LAYER_TYPE_MULTI_HEAD_ATTENTION` (since 7.0)       |                             |                             |                         | Multi head attention layer
+&#xfeff;                                                     | num_heads                   | (unsigned integer)          | 1                       | Number of head
+&#xfeff;                                                     | projected_key_dim           | (unsigned integer)          |                         | projected key dim per head
```
Comment (lhs8928) [inline docs/application/native/guides/machine-learning/machine-learning-training.md:278]: This is minor thing but can we start with capital letter? Cause other descriptions are also started with capital letter.
projected .. => Projected ...

Comment (songgot) [reply docs/application/native/guides/machine-learning/machine-learning-training.md]: thank you.

**Review by DonghakPark — APPROVED** [review]
LGTM

File: docs/application/native/guides/machine-learning/machine-learning-training.md [hunk 3]
```diff
@@ -266,8 +266,51 @@ Type | Key | Value | Default value | Description
...
+&#xfeff;                                                     |                             | lecun_normal                |                         | Lecun normal initialization
+&#xfeff;                                                     |                             | lecun_uniform               |                         | Lecun uniform initialization
+&#xfeff;                                                     |                             | xavier_normal               |                         | Xavier normal initialization
+&#xfeff;                                                     |                             | xavier_uniform              |                         | Xavier uniform initialization
+&#xfeff;                                                     |                             | he_normal                   |                         | He normal initialization
+&#xfeff;                                                     |                             | he_uniform                  |                         | he uniform initialization
```
Comment (safir-srbd) [inline docs/application/native/guides/machine-learning/machine-learning-training.md:308]:
```suggestion
&#xfeff;                                                     |                             | he_uniform                  |                         | He uniform initialization
```

**Review by safir-srbd — COMMENTED** [review]
@songgot Hello. Hope you are well.

I have provided a suggestion below. And for those parts where I couldn't suggest, here are the changes:

Line 455 - Following code is example of handling dataset:
Suggestion - **The** following code is **an** example of handling dataset:

Line 481 - As of now, only INI formatted files `*.ini` **is** supported to construct a model from a file.
Suggestion - As of now, only INI formatted files `*.ini` **are** supported to construct a model from a file.

Line 717 - The handle abstracts data providing logics with three dataset usage point:
Suggestion - The handle abstracts data, providing logics with three dataset usage point:

File: docs/application/native/guides/machine-learning/machine-learning-training.md (hunk 3 above)
Comment (songgot) [reply docs/application/native/guides/machine-learning/machine-learning-training.md]: thank you

File: docs/application/native/guides/machine-learning/machine-learning-training.md [hunk 4]
```diff
@@ -832,7 +875,4 @@ You can use the trained model with pipeline API as follows:
 ## Related information
 
 - Dependencies
-  - Tizen 6.0 and Higher for Mobile
-  - Tizen 6.0 and Higher for Wearable
-  - Tizen 6.0 and Higher for TV
-  - Tizen 6.0 and Higher for IoT
+  - Tizen 6.0 and Higher
```
Comment (safir-srbd) [inline docs/application/native/guides/machine-learning/machine-learning-training.md:878]:
```suggestion
  - Tizen 6.0 and Higher for Mobile
  - Tizen 6.0 and Higher for Wearable
  - Tizen 6.0 and Higher for TV
  - Tizen 6.0 and Higher for IoT
```
I think this format is more appropriate for Related information section in native articles.
Ref: https://github.com/Samsung/tizen-docs/blob/master/reviewguide/review_points_guide.md#dependencies

**Review by safir-srbd — COMMENTED** [review]
@songgot Thank you for implementing the changes. I have gone through the article again and found one more issue. Could you please address it?

File: docs/application/native/guides/machine-learning/machine-learning-training.md (hunk 4 above)
Comment (songgot) [reply docs/application/native/guides/machine-learning/machine-learning-training.md:878]: thank you.

**Review by safir-srbd — APPROVED** [review]
@songgot Thank you for making the suggested changes.

@asm-foysal @mijin85cho I am approving this PR for merging.

**Pull request merged**